libraries/AP_HAL_ChibiOS/RCOutput.cpp

```diff
@@ -134,7 +135,7 @@ void RCOutput::rcout_thread()
     // dshot is quite sensitive to timing, it's important to output pulses as
     // regularly as possible at the correct bitrate
     while (true) {
-        chEvtWaitOne(EVT_PWM_SEND | EVT_PWM_SYNTHETIC_SEND);
+        chEvtWaitOne(EVT_PWM_SEND | EVT_PWM_SYNTHETIC_SEND | EVT_LED_SEND);
```

Comment:
This will trigger dshot at irregular intervals and mess up the timing and counting. Why is this necessary? If you want to trigger an out of cycle LED update then you probably need to prevent the code below this treating it as a dshot cycle.
Note that the proper solution here (on my list) is to have a thread per timer group.

Reply:
its needed as without any dshot activity the thread gets stuck waiting for an event that never happens

Reply:
I've added a check for the PWM event bits for running the dshot code